### ISSUE TITLE
`Unknown` is overriding explicit type annotations

### ISSUE THREAD

**Issue by MarkusPielmeier** [astral-sh/ty#2064]
### Summary

A variable with explicit type annotation loses its type information when a value of type `Unknown` is assigned:

```py
from typing import reveal_type


# no type annotation to force ty inferring `Unknown`
def foo(): ...


def int_acceptor(value: int) -> None:
    print(value)


def str_acceptor(value: str) -> None:
    print(value)


def bar() -> None:
    x: int = foo()

    reveal_type(x)

    int_acceptor(x)
    str_acceptor(x)

```

In this example, i would expect the revealed type of `x` to be `int` due to its explicit annotation.
In consequence, i would expect ty to detect the type mismatch of calling `str_acceptor(x)`.

But actually, ty is revealing `x` being of type `Unknown` and no further diagnostics being issued.

``` txt
$ uvx ty check sandbox.py
info[revealed-type]: Revealed type
  --> sandbox.py:18:17
   |
17 |     x: int = foo()
18 |
19 |     reveal_type(x)
   |                 ^ `Unknown`
20 |
21 |     int_acceptor(x)
   |

Found 1 diagnostic
```

### Version

ty 0.0.3 (fadfe0966 2025-12-17)

**Comment by AlexWaygood** [comment]
Thanks for the report! Please see #136
